lib/dco.js

```diff
@@ -1,8 +1,28 @@
 module.exports = ({commit, parents}) => {
   const isMerge = parents && parents.length > 1;
-  return isMerge || commit.message.includes(signOff(commit));
+  const signed = commit.message.includes(signOff(commit));
+
+  if (isMerge || signed) {
+    return true;
+  } else if (!signed) {
+    return getError(commit);
+  }
 };
 
 function signOff(commit) {
   return `Signed-off-by: ${commit.author.name} <${commit.author.email}>`;
 }
+
+function getError(commit) {
+  let err = '';
+  if (!commit.message.includes('Signed-off-by:')) {
+    err = 'Your sign off is missing. ';
+  } else if (!commit.message.includes(commit.author.name) && !commit.message.includes(commit.author.email)) {
+    err = 'Neither author name nor email match. ';
+  } else if (!commit.message.includes(commit.author.name)) {
+    err = 'Your author name does not match. ';
+  } else if (!commit.message.includes(commit.author.email)) {
+    err = 'Your email does not match. ';
+  }
```

Comment:
It feels like there's some duplication between the `getError` method and the exported function at the top. I wonder if it could be reduced by replacing the `signOff` method with a regular expression that includes matching groups for the author name and email. Something like this (I haven't tested this, so will likely need tweaked): 

```js
const regex = /^Signed-off-by: (.*) <(.*)>$/gm
const [match, name, email] = commit.message.match(regex);

if (!match) {
  return 'The sign-off is missing.';
} else if (commit.author.name !== name || commit.author.email !== email) {
  return `The sign-off does not match the commit author: expected "${commit.author.name} <${commit.author.email}>", but got "${name} <${email}>"';
}
```